mlflow/pyfunc/__init__.py

```
@@ -845,10 +875,27 @@ def spark_udf(spark, model_uri, result_type="double"):
     # functionality.
     import functools
     from mlflow.pyfunc.spark_model_cache import SparkModelCache
+    from mlflow.utils._spark_utils import _SparkDirectoryDistributor
+    from mlflow.utils.nfs_on_spark import get_nfs_cache_root_dir
     from pyspark.sql.functions import pandas_udf
     from pyspark.sql.types import _parse_datatype_string
     from pyspark.sql.types import ArrayType, DataType as SparkDataType
     from pyspark.sql.types import DoubleType, IntegerType, FloatType, LongType, StringType
+    from pyspark.sql.functions import PandasUDFType
+    from mlflow.models.cli import _get_flavor_backend
+
+    # importing here to prevent circular import
+    from mlflow.pyfunc.scoring_server.client import prepare_env
+
+    if env_manager not in ["local", "conda"]:
+        raise ValueError("Illegal env_manager setting.")
```

Can we raise `MlflowException` with error code `INVALID_PARAMETER_VALUE`?